src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1544,6 +1549,17 @@ class ApplicationManagerImpl
   static std::vector<std::string> ConvertRejectedParamList(
       const std::vector<std::string>& input);
 
+  void AddExpiredButtonRequest(
+      const uint32_t app_id,
+      const int32_t corr_id,
+      const hmi_apis::Common_ButtonName::eType button_name) OVERRIDE;
+
+  bool IsButtonRequestExpired(const int32_t corr_id) const OVERRIDE;
```

Comment:
@ychernysheva I believe you can replace `IsButtonRequestExpired`/`GetAppIdFromExpiredButtonRequest`/`GetButtonNameFromExpiredButtonRequest` with a single method:
`utils::Optional<ExpiredButtonRequestData> GetExpiredButtonRequestData(const int32_t corr_id)`

Then you can simply check if optional is initialized then the request is expired. If it is expired, then you can use the values from optional to get both app_id and button_name

Reply:
As a sample, please take a look at 99a4c6dee779e2d63fef4bc3cfb3524c93510b7a